wolfcrypt/src/asn.c

```diff
@@ -35307,9 +35307,10 @@ enum {
     || (defined(HAVE_CURVE448) && defined(HAVE_CURVE448_KEY_IMPORT)) \
     || defined(HAVE_FALCON) || defined(HAVE_DILITHIUM) || defined(HAVE_SPHINCS))
 
+
 int DecodeAsymKey_Assign(const byte* input, word32* inOutIdx, word32 inSz,
     const byte** privKey, word32* privKeyLen,
-    const byte** pubKey, word32* pubKeyLen, int keyType)
+    const byte** pubKey, word32* pubKeyLen, int* inOutKeyType)
```

Comment:
Don't change the existing API!
This API is WOLFSSL_LOCAL which means customers may be using.

Make a new API DecodeAsymKey_Assign_ex() with these parameters and have DecodeAsymKey_Assign() call it.

Reply:
@SparkiDev  I'm getting pulled in multiple directions here. I originally did it just like you proposed in [f672105](https://github.com/wolfSSL/wolfssl/pull/8177/commits/f672105d557261eb376073faf06f12f347500ae3) but then @dgarske  reviewed and told me to just modify the original API instead of adding an `_ex` function, which I then changed in [2207791](https://github.com/wolfSSL/wolfssl/pull/8177/commits/2207791aab0a153ae13e250118c549817a33de7f). So I'm getting pulled in multiple directions here :frowning_face:. Can we please come to a consensus on how we want the API exposed and I will just do it that way? I don't care either way.

Reply:
OK.
Change the API then.